app/api/v2/resource/documents.rb

```diff
@@ -55,12 +60,13 @@ class Documents < Grape::API
           unless current_user.documents.count + params[:upload].length <= Barong::App.config.doc_num_limit
             error!({ errors: ['resource.documents.limit_will_be_reached'] }, 400)
           end
+          params[:identificator] = SecureRandom.hex(16) unless params[:identificator]
```

Comment:
We create identificator if its not passed in params, as we want all files linked in one request. (  :upload is an array of Rack::Multipart::UploadedFile)